homeassistant/components/group/notify.py

```diff
@@ -13,21 +16,30 @@
     PLATFORM_SCHEMA,
     BaseNotificationService,
 )
-from homeassistant.const import ATTR_SERVICE
+from homeassistant.components.switch import is_on
+from homeassistant.const import CONF_NAME, ATTR_SERVICE
 import homeassistant.helpers.config_validation as cv
 
 # mypy: allow-untyped-calls, allow-untyped-defs, no-check-untyped-defs
 
 _LOGGER = logging.getLogger(__name__)
 
 CONF_SERVICES = "services"
+CONF_SWITCH = "switch"
+CONF_MAX_PER_MINUTE = "max_per_minute"
+CONF_MAX_PER_HOUR = "max_per_hour"
+CONF_MAX_PER_DAY = "max_per_day"
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_SERVICES): vol.All(
             cv.ensure_list,
             [{vol.Required(ATTR_SERVICE): cv.slug, vol.Optional(ATTR_DATA): dict}],
-        )
+        ),
+        vol.Optional(CONF_SWITCH): cv.entity_id,
```

Comment:
Remove this option. This can be handled via an automation or similar.

Reply:
Thanks for the quick review & feedback, Martin!

The forums seem to have lots of suggestions about possibilities, but no one best way to do this.  Moreover, I can't think of a way that adds the abstraction that I'm aiming for here, which is a notification channel that I can turn on and off independently.  The alternatives I'm aware of are modifying every action to do this conditionally, and that's a very global change for such a simple intent.

What's the downside of the switch?  It's a clean abstraction that meshes with rate limiting (it's a temporary rate limiting to 0; one could imagine programmable/dynamicall-changeable rate limits, but that seemed like overkill compared to a simple switch).  And it's literally 4 lines of code with virtually zero runtime cost.

I hope you'll reconsider?

Reply:
Why should we have a switch for this condition? The condition could be anything that has a binary state. I don't think this is the correct abstraction level.

Reply:
I renamed the config setting to  'enable_entity_id' to make clear that this works with any entity_id that supports is_on() (which just happens to be declared in switch.py). Is that what you were looking for?

And please let me know if there's a more refined vol check to statically enforce the guarantee that is_on() will work.. I think all entities support that since the check is just whether the state is STATE_ON.